File: code/library/src/main/java/org/fbme/lib/iec61499/declarations/DeviceDeclaration.java
package org.fbme.lib.iec61499.declarations;

import org.fbme.lib.common.ContainedElement;
import org.fbme.lib.common.Declaration;
import org.fbme.lib.common.Reference;
import org.fbme.lib.iec61499.fbnetwork.FBNetwork;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface DeviceDeclaration extends Declaration, ContainedElement {

    @Nullable SystemDeclaration getContainer();

    @NotNull Reference<DeviceTypeDeclaration> getTypeReference();

    @NotNull List<ParameterAssignment> getParameters();

    @NotNull List<ResourceDeclaration> getResources();

    @Nullable FBNetwork getNetwork();
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/DeviceDeclaration.kt
package org.fbme.lib.iec61499.declarations

import org.fbme.lib.common.ContainedElement
import org.fbme.lib.common.Declaration
import org.fbme.lib.common.Reference
import org.fbme.lib.iec61499.fbnetwork.FBNetwork

interface DeviceDeclaration : Declaration, ContainedElement {
    override val container: SystemDeclaration?
    val typeReference: Reference<DeviceTypeDeclaration>
    val parameters: MutableList<ParameterAssignment>
    val resources: MutableList<ResourceDeclaration>
    val network: FBNetwork?
}
